### PR TITLE
Performance improvement

### PR DESCRIPTION
**Pull request by nctl144**
This PR aims to improve the performance of SCURL

### CODE REVIEW

**Comment by nctl144** [comment]
Hey @lopuhin , in this PR, I have successfully increased the performance of `urljoin` and `canonicalize_url` by doing the folowing:

+ Took care of the double parsing in `urljoin`
+ Removed unnecessary `to_native_str` call in `canonicalize_url` since we know the types of variables
+ made calls to our function `canonicalize_component` from chromium instead of calling `quote` from urllib.parse. 

These have brought significant increase to the performance of `canonicalize_url` as well as `urljoin`. In particular, the rate of link extraction from `canonicalize_url` increased from 31k links/sec to 44k links/sec. But let me know if there's anything that concerns you :)

**Comment by nctl144** [comment]
For some reason the cgurl.cpp is conflicted with the one from master branch...

File: benchmarks/canonicalize_test.py [hunk 1]
```diff
@@ -13,7 +13,7 @@ def main():
     time = 0
     time_canonicalize_url = 0
 
-    tar = tarfile.open("sites.tar.gz")
+    tar = tarfile.open("benchmarks/sites.tar.gz")
```
Comment (lopuhin) [inline benchmarks/canonicalize_test.py]: Completely unrelated, but do you recall why we don't we use the same chromium urls here as well?

File: scurl/cgurl.pyx [hunk 2]
```diff
@@ -85,7 +76,7 @@ cdef bytes unicode_handling(str):
         bytes_str = <bytes>str
     return bytes_str
```
Comment (lopuhin) [inline scurl/cgurl.pyx]: I'm curious how this works: we have ``cdef bytes bytes_str`` above and return it from the function, but return type of the function is ``char *`` - is this correct?

Comment (lopuhin) [reply scurl/cgurl.pyx]: Never mind, looks like Cython indeed does this conversion automatically.

**Comment by nctl144** [comment]
Hey @lopuhin , I have cleaned up the code even further by moving all the canonicalize code to `canonocalize.pyx` :) I think this PR is ready!

**Comment by nctl144** [comment]
Hey @lopuhin , yeah sorry about that... I did not notice yesterday 😄 I just fixed it now

**Pull request merged**